lib/views/home/survey_section/survey_list.dart

```diff
@@ -2,26 +2,100 @@ import 'package:flutter/material.dart';
 import 'package:kayla_flutter_ic/model/survey.dart';
 import 'package:kayla_flutter_ic/views/home/survey_section/survey_cell.dart';
 
+enum RefreshStyle {
+  swipeRightToRefresh,
+  pullDownToRefresh,
+}
+
+// ignore: must_be_immutable
 class SurveyList extends StatelessWidget {
+  final RefreshStyle refreshStyle;
   final List<Survey> surveyList;
   final PageController itemController;
   final ValueNotifier<int> onItemChange;
+  final Future<void> Function() onRefresh;
+  final Future<void> Function() onLoadMore;
+
+  final GlobalKey<RefreshIndicatorState> _refreshIndicatorKey =
+      GlobalKey<RefreshIndicatorState>();
+  bool _isLoading = false;
 
-  const SurveyList({
+  SurveyList({
     super.key,
+    required this.refreshStyle,
     required this.surveyList,
     required this.itemController,
     required this.onItemChange,
+    required this.onRefresh,
+    required this.onLoadMore,
   });
 
   @override
   Widget build(BuildContext context) {
-    return PageView.builder(
-      scrollDirection: Axis.horizontal,
-      controller: itemController,
-      onPageChanged: (index) => onItemChange.value = index,
-      itemCount: surveyList.length,
-      itemBuilder: (context, index) => SurveyCell(surveyList[index]),
+    _isLoading = false;
+    switch (refreshStyle) {
+      case RefreshStyle.swipeRightToRefresh:
+        return _buildSwipeRightToRefreshPageView(context);
+      case RefreshStyle.pullDownToRefresh:
+        return _buildPullDownToRefreshPageView(context);
+    }
+  }
+
+  Widget _buildSwipeRightToRefreshPageView(BuildContext context) {
+    final screenWidth = MediaQuery.of(context).size.width;
+    itemController.addListener(() {
+      if (itemController.offset < -10) {
+        _refreshIndicatorKey.currentState?.show();
+      }
+
+      if (itemController.offset > screenWidth * (surveyList.length - 1) + 10 &&
+          !_isLoading) {
+        _isLoading = true;
+        onLoadMore();
+      }
+    });
+    return RefreshIndicator(
+      key: _refreshIndicatorKey,
+      onRefresh: onRefresh,
+      child: PageView.builder(
+        scrollDirection: Axis.horizontal,
+        controller: itemController,
+        onPageChanged: (index) => onItemChange.value = index,
+        itemCount: surveyList.length,
+        itemBuilder: (context, index) => SurveyCell(surveyList[index]),
+      ),
+    );
+  }
+
+  Widget _buildPullDownToRefreshPageView(BuildContext context) {
+    final screenWidth = MediaQuery.of(context).size.width;
+    itemController.addListener(() {
+      if (itemController.offset > screenWidth * (surveyList.length - 1) + 10 &&
+          !_isLoading) {
```

Comment:
I think we can load more items if we swipe to near the end of the list (maybe 2-3 items remaining), it's more simple IMO 😄

Reply:
Changed to `- 3` 👍